### PR TITLE
Issue 280: Map Credentials does not work some times

### PR DESCRIPTION
**Pull request by santhosh-tekuri**
provider should handle each credential type separately

### CODE REVIEW

**Comment by santhosh-tekuri** [comment]
fixed the failure

File: core/src/main/java/denominator/Credentials.java [hunk 1]
```diff
@@ -111,6 +111,8 @@ public static Credentials from(Object... parts) {
      * @return copy of the credentials values
      * @throws IllegalArgumentException if input is not a {@code Map} or {@code List}, or it is
      *                                  empty.
+     * @deprecated This method is deprecated because it converts map credentials into list containing
+     * items in same order as in map. Providers should handle each type of credential separately
```
Comment (adriancole) [inline core/src/main/java/denominator/Credentials.java]: hypernit last sentence missing a period

File: ultradns/src/main/java/denominator/ultradns/UltraDNSTarget.java [hunk 2]
```diff
@@ -58,9 +59,25 @@ public String url() {
...
+      username = checkNotNull(mapCreds.get("username"), "username").toString();
+      password = checkNotNull(mapCreds.get("password"), "password").toString();
+    } else {
+      throw new IllegalArgumentException("Unsupported credential type: "+ creds);
+    }
+
     in.insert(0, url());
-    List<Object> creds = ListCredentials.asList(credentials.get());
-    in.body(format(SOAP_TEMPLATE, creds.get(0).toString(), creds.get(1).toString(),
+    in.body(format(SOAP_TEMPLATE, username, password,
```
Comment (adriancole) [inline ultradns/src/main/java/denominator/ultradns/UltraDNSTarget.java]: supernit: you can now bring `new String(in.body(), UTF_8)));` onto this line as it no longer needs to wrap

**Comment by adriancole** [comment]
A couple nits, but the code looks like the cure!

Please add the following to the changelog:
 * Allows map credentials to be used without regard to iteration order

Please also squash and reword your commit.

In feign and denominator, I'm trying to stop the behavior I started w/ not so hot commit messages. when this is complete, please squash your commits into one where the first line says what the commit does, and later references other issues. Ex.

Here's a nice commit message for the work you've done.
```
Allows map credentials to be used without regard to iteration order

ListCredentials.asList only works when an input map iterates in the same
order as Provider.credentialTypeToParameterNames(). This is a bad
assumption and the method is now deprecated.

closes #280
```
Note all lines wrap at 72 chars and summary doesn't wrap, nor has a period.
Note that we use `closes #` syntax to reference the issue, but not on the summary.

**Comment by adriancole** [comment]
LG I will update the CHANGELOG in a cherry-pick

**Comment by adriancole** [comment]
Thanks!

closed via e6dfb6c175385af900d3a4a012e342ab8030dc71

**Pull request closed without merging**